source/h1_connection.c

```diff
@@ -1512,36 +1533,42 @@ static int s_handler_process_read_message(
 
     (void)slot;
     struct aws_h1_connection *connection = handler->impl;
+    const size_t message_size = message->message_data.len;
 
     AWS_LOGF_TRACE(
-        AWS_LS_HTTP_CONNECTION,
-        "id=%p: Incoming message of size %zu.",
-        (void *)&connection->base,
-        message->message_data.len);
-
-    aws_linked_list_push_back(&connection->thread_data.read_buffer.messages, &message->queueing_handle);
+        AWS_LS_HTTP_CONNECTION, "id=%p: Incoming message of size %zu.", (void *)&connection->base, message_size);
 
-    if (s_try_process_read_messages(connection)) {
-        goto shutdown;
-    }
+    /* Shrink connection window by amount of data received */
+    AWS_ASSERT(message_size <= connection->thread_data.window_size);
```

Comment:
Is this actually assert-able? We can guarantee that this never happens?

Reply:
Yeah, the channel checks that `aws_channel_slot.window_size` will accommodate a message before allowing it to be sent.

The assert is because this is my personal calculation of window size, not the official window size, and I'm *fairly* certain my math is correct.

Reply:
Changed to actual error check